Rock.Encryption/CompositeCrypto.cs

```diff
@@ -158,14 +176,14 @@ private ICrypto GetDecryptCrypto(object keyIdentifier)
 
         private ICrypto GetCrypto(object keyIdentifier, Func<ICrypto, object, bool> canGet)
         {
-            var crypto = _cryptos.Where(c => canGet(c, keyIdentifier)).GetEnumerator();
+            var crypto = _cryptos.FirstOrDefault(c => canGet(c, keyIdentifier));
 
-            if (!crypto.MoveNext())
+            if (crypto == null)
             {
                 throw new KeyNotFoundException($"Unable to locate implementation of ICrypto that can locate a credential using keyIdentifier: {keyIdentifier}");
```

Comment:
Any place we though we should consider logging.

Reply:
I don't think so. The exception isn't in a background thread, so the user will see it.